src/backend/distributed/worker/worker_create_or_replace.c

```diff
@@ -75,32 +152,69 @@ WrapCreateOrReplace(const char *sql)
 Datum
 worker_create_or_replace_object(PG_FUNCTION_ARGS)
 {
-	text *sqlStatementText = PG_GETARG_TEXT_P(0);
-	const char *sqlStatement = text_to_cstring(sqlStatementText);
-	Node *parseTree = ParseTreeNode(sqlStatement);
+	List *sqlStatements = NIL;
 
 	/*
-	 * since going to the drop statement might require some resolving we will do a check
-	 * if the type actually exists instead of adding the IF EXISTS keyword to the
-	 * statement.
+	 * The first argument is either a text or a list of text's, we normalize this to a
+	 * list of statements for further use, where the variant with just a single text
+	 * argument will be treated as a list of statements with length 1
 	 */
-	ObjectAddress address = GetObjectAddressFromParseTree(parseTree, true);
-	if (ObjectExists(&address))
+	Oid argTypeOid = PG_GETARG_TYPE(0);
```

Comment:
as far as I know, PG handles the overloaded UDFs by using a different C function and then providing the `link_symbol `, see [here](https://www.postgresql.org/docs/14/sql-createfunction.html) :
```SQL
 CREATE OR REPLACE FUNCTION pg_catalog.worker_create_or_replace_object(statement text) RETURNS bool LANGUAGE C STRICT   AS 'citus', $$worker_create_or_replace_object_text$$;

 CREATE OR REPLACE FUNCTION pg_catalog.worker_create_or_replace_object(statement text[]) RETURNS bool LANGUAGE C STRICT   AS 'citus', $$worker_create_or_replace_object_array$$;
```

If we can use that approach, we can avoid `PG_GETARG_TYPE`

Reply:
I will have a look at that. It will require a bit more refactoring to not duplicate the internal logic too much. Having PG_GETARG_TYPE was a way to work around this. But I think you are right where it might be nicer to have a specialized function for the array variant.